cli/src/semdep/package_restrictions.py

```diff
@@ -33,14 +25,14 @@ def semver_matches(expression: str, actual_version: str) -> bool:
 
 # compare vulnerable range to version in lockfile
 def dependencies_range_match_any(
-    search_for_ranges: List[ProjectDependsOnEntry],
-    have_deps: List[LockfileDependency],
-) -> Iterator[Tuple[ProjectDependsOnEntry, LockfileDependency]]:
+    search_for_ranges: List[DependencyPattern],
+    have_deps: List[FoundDependency],
+) -> Iterator[Tuple[DependencyPattern, FoundDependency]]:
     for have_dep in have_deps:
         for target_range in search_for_ranges:
             if (
-                target_range.namespace.value.lower() == have_dep.namespace.value.lower()
-                and target_range.package_name == have_dep.name
+                target_range.ecosystem == have_dep.ecosystem
```

Comment:
Are we fine with not being always lower case?

Reply:
They're not strings anymore. They're classes from ATD. Agree that we should probably print them as lowercase though.

Reply:
Ok it's actually not easy to print them as lowercase in the json output because the json conversion function is generated by ATD and gets called deep into the conversion of a `CliMatch`

Reply:
can we just store the constant used by atd to be lower case? no reason Npm should ever have the N capitalized